### PR TITLE
feat: add default interval on period Input

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#80473

### CODE REVIEW

File: packages/core/src/components/templates/PeriodInput/PeriodInput.tsx [hunk 1]
```diff
@@ -90,24 +136,22 @@ const PeriodInput = ({
           nullable
           popup={horizontal}
           defaultDate={date}
-          onDateChange={_date => {
-            setDate(_date);
-            dateConfig.onDateChange(_date);
-          }}
+          onDateChange={_date => handleDateChange(mode, _date)}
```
Comment (vhu-axelor) [inline packages/core/src/components/templates/PeriodInput/PeriodInput.tsx:139]: Props mode is use to define isStartDate, instead directly send isStartDate

File: packages/core/src/components/templates/PeriodInput/PeriodInput.tsx [hunk 2]
```diff
@@ -49,12 +50,16 @@ const PeriodInput = ({
...
   style,
+  defaultInterval = null,
 }: PeriodInputProps) => {
   const I18n = useTranslator();
 
   const [startDate, setStartDate] = useState(startDateConfig.date);
   const [endDate, setEndDate] = useState(endDateConfig.date);
   const [isPeriodError, setIsPeriodError] = useState(false);
+  const [interval, setInterval] = useState(
+    defaultInterval !== null ? defaultInterval * 60 * 60 * 1000 : null,
```
Comment (vhu-axelor) [inline packages/core/src/components/templates/PeriodInput/PeriodInput.tsx:61]: Define a global const to transform hours to ms

File: packages/core/src/components/templates/PeriodInput/PeriodInput.tsx [hunk 3]
```diff
@@ -41,6 +41,7 @@ interface PeriodInputProps {
   showTitle?: boolean;
   horizontal?: boolean;
   style?: any;
+  defaultInterval?: number;
```
Comment (vhu-axelor) [inline packages/core/src/components/templates/PeriodInput/PeriodInput.tsx:44]: Maybe add "hours" to props name to make it clearer

File: changelogs/unreleased/80473.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "title": "PeriodInput: add defaultInterval to set a default interval between start and end dates",
```
Comment (vhu-axelor) [inline changelogs/unreleased/80473.json:2]: "PeriodInput: add interval in between start and end dates with possibility to define a default one in hours with props"

**Review by vhu-axelor — COMMENTED** [review]
When the component receives only one date and the default interval, the second date should be update

File: packages/core/src/components/templates/PeriodInput/PeriodInput.tsx [hunk 5]
```diff
@@ -64,6 +73,16 @@ const PeriodInput = ({
...
   }, [endDateConfig.date]);
 
+  useEffect(() => {
+    if (interval) {
+      if (startDate && !endDate) {
+        setEndDate(new Date(startDate.getTime() + interval));
+      } else if (endDate && !startDate) {
+        setStartDate(new Date(endDate.getTime() - interval));
+      }
+    }
```
Comment (lme-axelor) [inline packages/core/src/components/templates/PeriodInput/PeriodInput.tsx:83]: By doing this this way, the value outside is not updated. You are only updating the visual part but endDate is still empty where the component is used.

File: packages/core/src/components/templates/PeriodInput/PeriodInput.tsx [hunk 6]
```diff
@@ -23,6 +23,8 @@ import {DateInput} from '../../organisms';
 import {getEndOfDay, getStartOfDay} from '../../../utils';
 import {useTranslator} from '../../../i18n';
 
+const hoursToMilliseconds = (hours: number) => hours * 60 * 60 * 1000;
```
Comment (lme-axelor) [inline packages/core/src/components/templates/PeriodInput/PeriodInput.tsx:26]: Maybe we can move this to the time utils file in root utils folder as it is not linked to the component

**Review by vhu-axelor — APPROVED** [review]

**Pull request merged**